### PR TITLE
Support inheritance in plugin classes

### PR DESCRIPTION
**Pull request by tulinkry**
**This expects the #1525 to be merged as it already implements the new interface.**

I will rebase this when it gets merged.

This change is about:
 - support inheritance in plugin classes (currently the super class implementing the interface is not recognized)
 - do not necessarily create instances of abstract classes (because it does not work of course)

### CODE REVIEW

File: src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java [hunk 1]
```diff
@@ -415,30 +344,39 @@ private void handleLoadClass(String classname) {
...
         Class c = loader.loadClass(classname);
 
         // check for implemented interfaces
-        Class[] intf = c.getInterfaces();
-        boolean loaded = false;
-        for (Class intf1 : intf) {
-            if (intf1.getCanonicalName().equals(IAuthorizationPlugin.class.getCanonicalName())) {
+        for (Class intf1 : getInterfaces(c)) {
+            if (intf1.getCanonicalName().equals(IAuthorizationPlugin.class.getCanonicalName())
+                    && !Modifier.isAbstract(c.getModifiers())) {
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java]: this is one of the changes

File: src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java [hunk 2]
```diff
@@ -415,30 +344,39 @@ private void handleLoadClass(String classname) {
...
+     */
+    protected List<Class> getInterfaces(Class c) {
+        List<Class> interfaces = new LinkedList<>();
+        Class self = c;
+        while (self != null && !interfaces.contains(IAuthorizationPlugin.class)) {
+            interfaces.addAll(Arrays.asList(self.getInterfaces()));
+            self = self.getSuperclass();
         }
+        return interfaces;
     }
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java]: this is the other one

**Comment by tulinkry** [comment]
Now rebased

**Comment by vladak** [comment]
Looks reasonable, pls merge.

**Pull request merged**